protocol/src/main/java/com/amazon/opendistroforelasticsearch/sql/protocol/response/format/FlatResponseFormatter.java

```diff
@@ -0,0 +1,126 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.sql.protocol.response.format;
+
+import com.amazon.opendistroforelasticsearch.sql.protocol.response.QueryResult;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@RequiredArgsConstructor
+public abstract class FlatResponseFormatter implements ResponseFormatter<QueryResult> {
```

Comment:
Could you make `CSVResponseFormatter` subclass? Otherwise there are many duplicated code in both class.

Reply:
@dai-chen let me make CSVResponseFormatter and address your other comments as well, thanks